sources/logdog/src/main.rs

```diff
@@ -0,0 +1,221 @@
+/*!
+# Introduction
+
+`logdog` is a program that gathers logs from various places on a Bottlerocket host and combines them
+into a tarball for easy export.
+
+Usage example:
+```
+$ logdog
+logs are at: /tmp/bottlerocket-logs.tar.gz
+```
+
+# Logs
+
+For the commands used to gather logs, please see [log_request](src/log_request.rs).
+
+*/
+
+#![deny(rust_2018_idioms)]
+
+mod create_tarball;
+mod error;
+mod log_request;
+
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::{env, process};
+
+use create_tarball::create_tarball;
+use error::Result;
+use log_request::{log_requests, LogRequest};
+use shell_words;
```

Comment:
```suggestion
```

Similar to tar, you shouldn't need to `use` a crate name; your dependencies are already in scope.  `use` is there to make names easier, not to make them possible, if that makes sense.  (with few exceptions now in Rust 2018)